Task 03

denisgubin opened the pull request:
решение заданий модуля 03
1
2
3
4

denisgubin committed:
task 02-sort-strings
task-02-pick
task-02-omit

denisgubin committed:
task 02-sort-strings
task-02-pick
task-02-omit

избавление от лишнего кода ниже:

let regExpEn = /[a-zA-z]/;
let regExpRu = /[а-яА-ЯЁё]/;

  let firstCharacterOfFirstElem = copyArray[0].charAt(0);

  let locale = getLocale(firstCharacterOfFirstElem);
  if (locale === "") {
    return arr;
  }

function getLocale(word) {
  let matchEn = regExpEn.test(word);
  if (matchEn === true) {
    return "en-US"; // determined 'en-US' locale
  }

  let matchRu = regExpRu.test(word);
  if (matchRu === true) {
    return "ru-RU"; // determined 'ru-RU' locale
  }

  return "";
}

/**
 * checkInList - Checking either value exists in fields
 * @param {string} value - value to find
 * @param {...string} fields - properties
 * @returns {boolean} - true if value exists in fields else false
 */
export function checkInList(value, ...fields) {
  for (let field of fields) {
    if (value === field) {
      return true;
    }
  }

  return false;
}

denisgubin committed:
task 02-sort-strings
task-02-pick
task-02-omit

избавление от лишнего кода ниже:

/**
 * checkInList - Checking either value exists in fields
 * @param {string} value - value to find
 * @param {...string} fields - properties
 * @returns {boolean} - true if value exists in fields else false
 */
export function checkInList(value, ...fields) {
  for (let field of fields) {
    if (value === field) {
      return true;
    }
  }

  return false;
}

denisgubin committed:
Merge branch 'master' of https://github.com/js-tasks-ru/js-20230914_denis-gubin into task_03

denisgubin committed:
Merge branch 'master' of https://github.com/js-tasks-ru/js-20230914_denis-gubin into task_03

denisgubin committed:
Merge branch 'master' of https://github.com/js-tasks-ru/js-20230914_denis-gubin into task_03

denisgubin committed:
Merge branch 'master' of https://github.com/js-tasks-ru/js-20230914_denis-gubin into task_03

denisgubin committed:
task3-01
task3-02
task3-03
task3-04

jsru commented:
Добавляю преподавателя (@ufocoder) для код-ревью.

denisgubin committed:
task02 задача 01
Правки по функции sortStrings

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

denisgubin committed:
task02 задача 01
Правки по функции sortStrings

jsru commented:
Добавляю преподавателя (@ufocoder) для код-ревью.

denisgubin committed:
Merge branch 'master' of github.com:denisgubin/js-20230914_denis-gubin into task_03

jsru commented:
Добавляю преподавателя (@ufocoder) для код-ревью.

denisgubin committed:
task02 задача 01
Правки по функции sortStrings

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

denisgubin committed:
task02 задача 01
Вернулся к старому решению

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

denisgubin committed:
task02 задача 01
Опять новое решение

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

ufocoder commented on 02-javascript-data-types/1-sort-strings/index.js:
```diff
@@ -4,6 +4,25 @@
+const localeRu = "ru-RU";
+
 export function sortStrings(arr, param = 'asc') {
+  let copyArray = [...arr];
```
В чем необходимость `let` если переменная не перезаписывается? Используйте `const`

ufocoder commented on 02-javascript-data-types/1-sort-strings/index.js:
```diff
@@ -4,6 +4,25 @@
 
+  return copyArray.sort(compare(param));
+}
+// можно ли в js обходится без стрелочных функций, это же не будет not js style ?
```
Не задавайте вопросы в виде комментариев, есть другие способы коммуникации

ufocoder commented on 03-objects-arrays-intro-to-testing/1-create-getter/index.js:
```diff
@@ -4,5 +4,19 @@
+      return obj;
+    }
+
+    let keys = Object.keys(obj);
```
`const`

ufocoder commented on 03-objects-arrays-intro-to-testing/4-uniq/index.js:
```diff
@@ -4,5 +4,19 @@
+  }
 
+  let uniqArr = [];
+  let tmpMap = new Map();
```
Попробуйте для этой задачи использовать объект `Set`

ufocoder commented on 02-javascript-data-types/1-sort-strings/index.js:
```diff
@@ -4,6 +4,25 @@
+// можно ли в js обходится без стрелочных функций, это же не будет not js style ?
+function compare(setDirection) {
+  if (setDirection === 'desc') {
+    return function (a, b) {
```
Выделите эти две функции в отдельные, за функцией compare (по желанию)

denisgubin committed:
task02 и task03
правка задач

denisgubin committed:
Merge branch 'master' of github.com:denisgubin/js-20230914_denis-gubin into task_03

# Conflicts:
#	02-javascript-data-types/1-sort-strings/index.js

jsru commented:
Добавляю преподавателя (@ufocoder) для код-ревью.

ufocoder commented on 03-objects-arrays-intro-to-testing/2-invert-object/index.js:
```diff
@@ -4,5 +4,15 @@
  */
 export function invertObj(obj) {
+  if (obj === undefined) {
+    return undefined;
```
`return undefined;` -> `return`

ufocoder commented on 03-objects-arrays-intro-to-testing/1-create-getter/index.js:
```diff
@@ -4,5 +4,19 @@
+
+    let keys = Object.keys(obj);
+    for (let i = 0; i < keys.length;i++) {
+      if (keys[i] === partParts[i]) {
```
Попробуйте уйти от работы с циклом по возможности, код не очевиден

denisgubin committed:
правка задачи Task03-1

/*
можно было бы сделать так
deepIntoObj(partParts.shift());

но не очень хорошо, когда из массива удаляется 1-й элемент, т.к. все элементы масссива должены сместиться на 1-н назад.
 */

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

denisgubin committed:
правка задачи Task03-1

по ошибке в коде оставил запись deepIntoObj(partParts.shift());

jsru commented:
Добавляю преподавателя (@ufocoder) для код-ревью.

ufocoder commented on 03-objects-arrays-intro-to-testing/1-create-getter/index.js:
```diff
@@ -4,5 +4,21 @@
  * @returns {function} - function-getter which allow get value from object by set path
  */
 export function createGetter(path) {
```
Попробуйте проверить функцию на этих примерах, у меня выдавало некорректный результат:
```
const getter = createGetter('a.b')

getter({ a: null });
getter({ a: { b: { c: {} }}});
getter({ a: { b: { c: null }}});
```
и попробуйте еще так:
```
const getter = createGetter('a.b')

getter({ a: { b: 42 }});
getter({ a: { b: 42 }}); // у вас не сбрасывается currentIndex - возможно проблема с повторным использованием
```

denisgubin committed:
- правка задачи Task03-1:
- добавлено сбрасывание счётчика
- добавлена проверка значения объекта на null и undefined

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).
